code/game/gamemodes/events/powerfruit.dm

```diff
@@ -0,0 +1,90 @@
+#define POWER_PER_FRUIT 500
+#define SPREAD_CHANCE 15
+#define ATTACK_CHANCE 15
+
+//the actual powerfruit obj
+/obj/structure/cable/powerfruit
+	name = "powerfruit"
+	desc = "A strange alien fruit that passively generates electricity. Best not to touch it."
+	icon = 'icons/obj/lighting.dmi' //TODO
+	icon_state = "glowshroomf" //TODO
+	level = LEVEL_ABOVE_FLOOR
+	plane = ABOVE_HUMAN_PLANE
+	pass_flags = PASSTABLE | PASSGRILLE | PASSGIRDER | PASSMACHINE
+
+/obj/structure/cable/powerfruit/New(loc)
+	var/datum/powernet/PN = getFromPool(/datum/powernet)
+	PN.add_cable(src)
+
+	for(var/dir in cardinal)
+		mergeConnectedNetworks(dir)   //Merge the powernet with adjacents powernets
+	mergeConnectedNetworksOnTurf() //Merge the powernet with on turf powernets
+
+	processing_objects += src
+	. = ..()
+
+/obj/structure/cable/powerfruit/Destroy()
+	processing_objects -= src
+	..()
+
+/obj/structure/cable/powerfruit/process()
+	//add power to powernet
+	add_avail(POWER_PER_FRUIT)
+
+	//spread - copypasta from spreading_growth.dm
+	if(prob(SPREAD_CHANCE))
+		var/list/neighbors = getViableNeighbours()
+		sleep(rand(3,5))
+		if(!(gcDestroyed || !neighbors.len))
+			var/turf/target_turf = pick(neighbors)
+			var/obj/structure/cable/powerfruit/child = new(get_turf(src))
+			neighbors -= target_turf
+			spawn(1) // This should do a little bit of animation.
+				child.forceMove(target_turf)
+
+	//if there is a person caught in the vines, burn em a bit
+	//electrocute people who aren't insulated
+	for(var/mob/living/M in range(1, get_turf(src)))
+		if(prob(ATTACK_CHANCE))
```

Comment:
Rather than generating and populating a list that may not do something, instead have it check for probability, THEN look for living in range to electrocute.

Reply:
this is so the chance is run for every person, not once for all